src/core/unit.c

```diff
@@ -4011,6 +4011,8 @@ void unit_warn_if_dir_nonempty(Unit *u, const char* where) {
         assert(u);
         assert(where);
 
+        if (access(where, F_OK) == 0)
+            return;
```

Comment:
coding style: please indent by multiples of 8 spaces, see CODING_STYLE...

More importantly though, what are you trying to do here? the above checks whether the specified path exists at all with no check what kind of file node this actually is, i.e. a directory, a regular file or anything else...

I am not sure I fully grok what you are trying to fix here: is it that for bind mounts regular files are fine as node to mount over, and hence we shouldn't generate a warning in that case? If so, it sounds more appropriateto leave the dir_is_empty() as is, but then check explicitly look for the -ENOTDIR error it might return, and if so simply return quickly without complaining.

Reply:
I missed the spaces even though I did go through the coding style document. Sorry.

Back to the problem at hand. When using mount units to bind mount files systemd will complain saying something like:
`foo.mount: Failed to check directory /foo: Not a directory`
This is because `unit_warn_if_dir_nonempty` is tried every time assuming we are always mounting a directory outputting an irrelevant warning if that is not true.

I like the solution with -ENOTDIR. I'll resend.